core/tauri-macros/src/lib.rs

```diff
@@ -151,3 +154,38 @@ pub fn do_menu_item(input: TokenStream) -> TokenStream {
   let tokens = parse_macro_input!(input as menu::DoMenuItemInput);
   menu::do_menu_item(tokens).into()
 }
+
+/// Convert a .png or .ico icon to an Image
+/// for things like `tauri::tray::TrayIconBuilder` to consume,
+/// relative paths are resolved from `CARGO_MANIFEST_DIR`, not current file
+///
+/// ### Examples
+///
+/// ```ignore
+/// TrayIconBuilder::new().icon(include_image!("./icons/32x32.png")).build().unwrap();
+/// ```
+///
+/// Note: this stores the image in raw pixels to the final binary,
+/// so keep the icon size (width and height) small
+/// or else it's going to bloat your final executable
+#[proc_macro]
+pub fn include_image(tokens: TokenStream) -> TokenStream {
+  let Ok(path) = parse2::<LitStr>(tokens.into()) else {
+    return quote!(compile_error!("Macro body is not a string")).into();
+  };
```

Comment:
you can just use the built-in compile_error from syn crate
```suggestion
  let path = match parse2::<LitStr>(tokens.into()) {
    Ok(path) => path,
    Err(err) => return err.into_compile_error().into();
  };
```